src/h5web/visualizations/heatmap/HeatmapVis.tsx

```diff
@@ -16,9 +17,12 @@ function HeatmapVis(props: Props): JSX.Element {
   const { dims, data } = props;
   const values = data.flat();
 
-  const interpolator = interpolateMagma;
   const domain = findDomain(values);
-  const textureData = computeTextureData(values, domain, interpolator);
+  const [colorMap, setColorMap] = useState<ColorMap>({
+    name: 'Magma',
+    interpolator: interpolateMagma,
+  });
```

Comment:
You could probably simplify a few things by not having a `ColorMap` interface. That is, instead of keeping both the name of the colour map and the intepolator in a state here, you could just store an enum value, like `Interpolator.Magma`. The enum would live in `./models.ts`.

The record that you use to map colour map names to interpolators in `ColorMapSelector`, could then be moved to `utils` and you could use it directly to access the interpolator from the enum any time you need it (like below).

Reply:
I now have a single record with names in keys and interpolators as values. The `ColorMap` type models variables that are the keys of this record.